impc_prod_tracker/src/main/java/uk/ac/ebi/impc_prod_tracker/conf/security/AapSystemSubject.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *******************************************************************************/
+package uk.ac.ebi.impc_prod_tracker.conf.security;
+
+import io.jsonwebtoken.Claims;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.impc_prod_tracker.domain.ProductionCentre;
+import uk.ac.ebi.impc_prod_tracker.domain.UserRole;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@Data
+@NoArgsConstructor
+/**
+ * Implementation of SystemSubject where most of the user information is taken from a token (jwt). Additional
+ * information needs to be loaded from the database.
+ * @author Mauricio Martinez
+ */
+public class AapSystemSubject implements SystemSubject
```

Comment:
General style comment -- prefer the javadoc above the annotations as the annotations effect the code directly so should be closer.  See: https://docs.oracle.com/javase/1.5.0/docs/guide/javadoc/deprecation/deprecation.html

Reply:
Done.